src/Functions/reinterpretAs.cpp

```diff
@@ -255,16 +268,45 @@ class FunctionReinterpret : public IFunction
 
                     static constexpr size_t copy_size = std::min(sizeof(From), sizeof(To));
 
-                    for (size_t i = 0; i < size; ++i)
+                    if constexpr (std::is_same_v<FromType, DataTypeUUID>)
                     {
-                        if constexpr (std::endian::native == std::endian::little)
-                            memcpy(static_cast<void*>(&to[i]), static_cast<const void*>(&from[i]), copy_size);
-                        else
+                        size_t i = 0;
```

Comment:
I've rewritten it a little to remove the use of the `std::views::transform`. It would be nice to keep the two `for_each` calls on L270 and L375, as I need to modify a copy of the data in-place. The usage of a lambda function here affords this.